x-pack/platform/plugins/private/product_intercept/public/plugin.ts

```diff
@@ -9,153 +9,196 @@ import { Subscription } from 'rxjs';
 import React from 'react';
 import { i18n } from '@kbn/i18n';
 import { EuiLink, EuiText } from '@elastic/eui';
-import { CoreSetup, CoreStart, Plugin } from '@kbn/core/public';
+import { type CoreSetup, type CoreStart, Plugin } from '@kbn/core/public';
+import type { PluginInitializerContext } from '@kbn/core/public';
 import { InterceptsStart } from '@kbn/intercepts-plugin/public';
+import { type CloudStart } from '@kbn/cloud-plugin/public';
 import { FormattedMessage } from '@kbn/i18n-react';
 
 import { NPSScoreInput } from './components';
 import { PromptTelemetry } from './telemetry';
-import { TRIGGER_DEF_ID } from '../common/constants';
+import { TRIGGER_DEF_ID, UPGRADE_TRIGGER_DEF_PREFIX_ID } from '../common/constants';
 
 interface ProductInterceptPluginStartDeps {
   intercepts: InterceptsStart;
+  cloud: CloudStart;
 }
 
 export class ProductInterceptPublicPlugin implements Plugin {
   private readonly telemetry = new PromptTelemetry();
   private interceptSubscription?: Subscription;
+  private upgradeInterceptSubscription?: Subscription;
+  private readonly buildVersion: string;
+
+  constructor(ctx: PluginInitializerContext) {
+    this.buildVersion = ctx.env.packageInfo.version;
+  }
 
   setup(core: CoreSetup) {
     return this.telemetry.setup({ analytics: core.analytics });
   }
 
-  start(core: CoreStart, { intercepts }: ProductInterceptPluginStartDeps) {
+  start(core: CoreStart, { intercepts, cloud }: ProductInterceptPluginStartDeps) {
     const eventReporter = this.telemetry.start({
       analytics: core.analytics,
     });
 
-    this.interceptSubscription = intercepts
-      .registerIntercept?.({
-        id: TRIGGER_DEF_ID,
-        steps: [
-          {
-            id: 'start',
-            title: i18n.translate('productIntercept.prompter.step.start.title', {
-              defaultMessage: 'Help us improve Kibana',
-            }),
-            content: () =>
-              React.createElement(
-                EuiText,
-                { key: 'productInterceptPrompterStartContent', size: 's' },
-                i18n.translate('productIntercept.prompter.step.start.content', {
+    const productOffering = `Elastic ${function (string: string) {
+      return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
+    }.call(null, cloud.serverless?.projectType || '')}`.trim();
+
+    void (async () => {
+      const currentUser = await core.security.authc.getCurrentUser();
+
+      const surveyUrl = new URL('https://elastic.eu.qualtrics.com/jfe/form/SV_4OfUp18saHW14Mu');
```

Comment:
Sorry if this has been mentioned before, but this better be Elastic's short URL so we can fix it if it turns into a pumpkin. https://links.elastic.dev

Reply:
I checked that it seems to be passing the search parameters forward.